great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py

```diff
@@ -55,12 +107,200 @@ def kwargs(self) -> dict:
+
+        Given a list of terms created by parsing a provided condition, recursively substitute all parameters and
+        variables in the term list, regardless of depth of groupings.
+
+        Example:
+            condition: "($variables.max_user_id>0 & $variables.answer==42) | $parameter.my_min_user_id.value[0]<0" will
+            return the following term list from self._parse_condition:
+                parsed_condition = [[
+                                      [
+                                        ['$variables.max_user_id', '>', 0],
+                                        '&',
+                                        ['$variables.answer', '==', 42]
+                                      ],
+                                      '|',
+                                      ['$parameter.my_min_user_id.value[0]', '<', 0]
+                                   ]]
+
+            This method will then take that term list and recursively search for parameters and variables that need to
+            be substituted and return this ParseResults object:
+                return [[
+                          [
+                             [999999999999, '>', 0],
+                             '&',
+                             [42, '==', 42]
+                          ],
+                          '|',
+                          [397433, '<', 0]
+                       ]]
+
+        Args:
+            term_list (Union[str, ParseResults): the ParseResults object returned from self._parse_condition
+            domain (Domain): The domain of the ExpectationConfiguration
+            variables (Optional[ParameterContainer]): The variables set for this ExpectationConfiguration
+            parameters (Optional[Dict[str, ParameterContainer]]): The parameters set for this ExpectationConfiguration
+
+        Returns:
+            ParseResults: a ParseResults object identical to the one returned by self._parse_condition except with
+                          substituted parameters and variables.
+        """
+        idx: int
+        token: Union[str, ParseResults]
+        for idx, token in enumerate(term_list):
+            if isinstance(token, str) and token.startswith("$"):
+                term_list[idx]: Dict[str, Any] = get_parameter_value(
+                    domain=domain,
+                    parameter_reference=token,
+                    variables=variables,
+                    parameters=parameters,
+                )
+            elif isinstance(token, ParseResults):
+                self._substitute_parameters_and_variables(
+                    term_list=token,
+                    domain=domain,
+                    variables=variables,
+                    parameters=parameters,
+                )
+        return term_list
+
+    def _build_bitwise_list(
+        self,
+        substituted_term_list: Union[str, ParseResults],
+    ) -> ParseResults:
+        """Recursively build bitwise list from substituted terms
```

Comment:
@NathanFarmer What does "bit-wise" mean in this context?  Is it regarding manipulating binary numbers, or dealing with logical operators that use two operands?  Or a different meaning?  Thanks!